tle/cogs/codeforces.py

```diff
@@ -33,13 +34,28 @@ def __init__(self, bot):
         rc = cf_common.user_db.update_status(active_ids)
         await ctx.send(f'{rc} members active with handle')
 
-    async def _gitgud(self, ctx, handle, problem, delta):
+    async def _validate_gitgud_status(self, ctx, delta):
+        if delta % 100 != 0:
+            await ctx.send('Delta must be a multiple of 100.')
+            return False
+
+        if abs(delta) > _GITGUD_MAX_ABS_DELTA_VALUE:
+            await ctx.send(f'Delta must range from -{_GITGUD_MAX_ABS_DELTA_VALUE} to {_GITGUD_MAX_ABS_DELTA_VALUE}.')
+            return False
+
         user_id = ctx.message.author.id
         active = cf_common.user_db.check_challenge(user_id)
         if active is not None:
             challenge_id, issue_time, name, contest_id, index, c_delta = active
             url = f'{cf.CONTEST_BASE_URL}{contest_id}/problem/{index}'
             await ctx.send(f'You have an active challenge {name} at {url}')
+            return False
+
+        return True
+
+    async def _gitgud(self, ctx, handle, problem, delta):
+        user_id = ctx.message.author.id
+        if not await self._validate_gitgud_status(ctx, delta):
```

Comment:
This is a wrong spot to make this check. _gitgud is called at the very end, to create a new challenge. You should make this check on top of ;gitgud (which you did) and on top of ;upsolve (which also creates a challenge)